flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/simple/PersistentKafkaSource.java

```diff
@@ -211,6 +215,16 @@ public void run(Collector<OUT> collector) throws Exception {
 		}
 	}
 
+	public void setConnectTimeoutMs(int connectTimeoutMs) {
+		Preconditions.checkArgument(connectTimeoutMs > 0, "The timeout must be positive");
+		this.connectTimeoutMs = connectTimeoutMs;
+	}
+
+	public void setBufferSize(int bufferSize) {
+		Preconditions.checkArgument(connectTimeoutMs > 0, "The buffer size be positive");
```

Comment:
typo: missing "must"... I think we should add these checks in the constructor as well.